course02/homework01/gavin/list.cpp

```diff
@@ -0,0 +1,143 @@
+#include <memory>
+
+#include <memory>
+
+#include "list.h"
+
+void List::clear() {
+    std::unique_ptr<Element> newList;
+    mFirst = std::move(newList);
+    mSize = 0;
+}
+
+void List::insert(std::size_t pos, int data)
+{
+    if(pos == 0)
+    {
+        push_front(data);
+        return;
+    }
+    if(pos == size())
+    {
+        push_back(data);
+        return;
+    }
```

Comment:
To get more elegant code, I would have imagine the calls the other way around: that `push_front` would call `insert(0)` and `push_back` `insert(size())` !